experimental/framework/tree-react-api/src/useTree.ts

```diff
@@ -3,30 +3,43 @@
  * Licensed under the MIT License.
  */
 
-import { ISharedTreeView } from "@fluid-experimental/tree2";
+import {
+	GlobalFieldSchema,
+	ISharedTreeView,
+	SchemaAware,
+	SchematizeConfiguration,
+} from "@fluid-experimental/tree2";
 import React from "react";
 
 /**
- * Returns the root of the given ISharedTree instance.  The returned object tree may
- * be mutated and passed as properties to subcomponents.
+ * Returns the root of the given ISharedTree instance.
+ * The returned object tree may be mutated and passed as properties to sub-components.
  *
- * Be aware that object retained between render passes will mutate.  Use deep clone
- * if saving a copy for comparison.
+ * Be aware that object retained between render passes will mutate.
+ * Use deep clone if saving a copy for comparison.
  *
  * Not currently compatible with 'React.memo'.
  */
-export function useTree<T>(tree: ISharedTreeView): T {
+export function useTree<TRoot extends GlobalFieldSchema>(
+	tree: ISharedTreeView,
+	config: SchematizeConfiguration<TRoot>,
+): SchemaAware.TypedField<SchemaAware.ApiMode.Editable, TRoot["schema"]> {
```

Comment:
Seems odd to re-schematize the tree on each render?

Reply:
callers of this function were using it to do the type conversion, which is what schematize does. CHanging that needs some refactroing of these examples.

Also, doing schematize here (though likley in a memo or something) might lead to easier support for error cases when a remote collaborator edits ther scehrma to an incompatible one.

I have put it in a React.memo and added a todo to revisite this.